pkg/option/config.go

```diff
@@ -116,6 +122,16 @@ type daemonConfig struct {
 	// This variable should never be written to, it is initialized via
 	// daemonConfig.Validate()
 	IPv6ClusterAllocCIDRBase string
+
+	// K8sRequireIPv4PodCIDR requires the k8s node resource to specify the
+	// IPv4 PodCIDR. Cilium will block bootstrapping until the information
+	// is available.
+	K8sRequireIPv4PodCIDR bool
```

Comment:
Any concern about how this may affect our readiness probes?

Reply:
The readinessProbe happens every 5 seconds so not a problem. The livenessProbe has a delay of 2 minutes, if the PodCIDR is not announced within that timeframe, the cilium pod will declared unhealthy and restarted. I think that's what we want.